app/src/main/java/net/sourceforge/opencamera/ExtendedAppInterface.java

```diff
@@ -1,78 +1,165 @@
 package net.sourceforge.opencamera;
 
+import android.app.AlertDialog;
+import android.content.BroadcastReceiver;
+import android.content.Context;
 import android.content.Intent;
+import android.content.IntentFilter;
 import android.content.SharedPreferences;
 import android.hardware.Sensor;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
 import android.net.wifi.WifiManager;
+import android.os.AsyncTask;
+import android.os.Build;
 import android.os.Bundle;
+import android.os.Handler;
 import android.preference.PreferenceManager;
 import android.util.Log;
+import android.view.View;
+import android.widget.ImageButton;
 import android.widget.TextView;
 
+import androidx.annotation.RequiresApi;
+
+import com.googleresearch.capturesync.PhaseAlignController;
 import com.googleresearch.capturesync.SoftwareSyncController;
+import com.googleresearch.capturesync.softwaresync.SoftwareSyncLeader;
+import com.googleresearch.capturesync.softwaresync.phasealign.PeriodCalculator;
+import com.googleresearch.capturesync.softwaresync.phasealign.PhaseConfig;
 
+import net.sourceforge.opencamera.cameracontroller.CameraController;
 import net.sourceforge.opencamera.cameracontroller.YuvImageUtils;
+import net.sourceforge.opencamera.preview.Preview;
 import net.sourceforge.opencamera.sensorlogging.FlashController;
 import net.sourceforge.opencamera.sensorlogging.RawSensorInfo;
 import net.sourceforge.opencamera.sensorlogging.VideoFrameInfo;
 import net.sourceforge.opencamera.sensorlogging.VideoPhaseInfo;
+import net.sourceforge.opencamera.ui.MainUI;
+import net.sourceforge.opencamera.ui.ManualSeekbars;
+
+import org.json.JSONException;
+import org.json.JSONObject;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Field;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Date;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.concurrent.BlockingQueue;
 
 /**
- * Extended implementation of ApplicationInterface, adds raw sensor recording layer to the
- * interface.
+ * Extended implementation of ApplicationInterface, adds raw sensor recording layer and RecSync
+ * controls to the interface.
  */
 public class ExtendedAppInterface extends MyApplicationInterface {
```

Comment:
to prevent huge deltas with open cmaera code you can follow next pattern. Extract all your code to some class for example SensorsExtendedAppInterface. Construct it inside ExtendedAppInterface and in all necessary callbacks invoe same callbacks in your class. If you need fields of ExtendedAppInterface use constructor injecting for never changed fields or custom setters which inform your class about changes for changing fields

Reply:
If I understand you correctly, `ExtendedAppInterface` is exactly what you are talking about: it is the extension of `MyApplicationInterface`. And `MyApplicationInterface` is present in the original OpenCamera code, while `ExtendedAppInterface` is not, so it the class that we use to prevent the huge deltas you are talking about.

Reply:
Okey, it sounds great. I can only mention that size of this file is really huge, but in my opinion it isn't critical now